src/features/workflow/state/workflowState.ts

```diff
@@ -289,6 +292,17 @@ function moveWorkflowTransformStepUp (state: WorkflowState, action: WorkflowStep
   return
 }
 
+function workflowUndoChanges (state: WorkflowState, actions: UndoWorkflowChanges) {
+  const oldTriggers:WorkflowTrigger[] | undefined = JSON.parse(JSON.stringify(state.workflowBase.triggers))
+  const oldHooks: WorkflowHook[] | undefined = JSON.parse(JSON.stringify(state.workflowBase.hooks))
+  const oldSteps: TransformStep[] = JSON.parse(JSON.stringify(state.workflowBase.steps || []))
```

Comment:
why do we need this double JSON parse/unparse?

Reply:
its a trick to make a deep copy of an object